rotary_utils/repair.py

```diff
@@ -158,27 +158,50 @@ def __init__(self, linking_outdir: str, length_threshold: int = None):
         """
         self.linking_outdir = linking_outdir
         self.log_dir_base = os.path.join(linking_outdir, 'logs')
-        self.length_threshold = length_threshold
+        self._current_length_threshold = length_threshold
 
-        if length_threshold:
-            self.length_outdir = os.path.join(linking_outdir, f'L{length_threshold}')
-            self.log_dir = os.path.join(self.log_dir_base, f'L{length_threshold}')
-        else:
-            self.length_outdir = None
-            self.log_dir = None
+    @property
```

Comment:
Methods (`.length_threshold(X)`) that we want to show up as attributes (`.length_threshold = X`) are labelled with the `@property` decorator.